specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.4-preview.1/settings.json

```diff
@@ -188,7 +188,7 @@
     },
     "SettingsListResult": {
```

Comment:
Anything returning a `SettingsListResult` should be marked as pageable (see x-ms-pageable).

Reply:
@heaths I think we purposely did not make it pageable for dev experience, since there are only few settings without any plans to grow above threshold in the future.

Reply:
Yes, and I communicated this as well. But I made the same arguments to the service team before that marking this as pageable now provides flexibility. Customers could only fetch the first page of results, but I know some languages make this easy anyway. In .NET, you can `foreach` through all items in all pages (if more than one page) or `await foreach` for an async pageable. I believe other languages are similar, if not more intuitive.

At the same time, @johanste, could this be discussed more and made for GA? I'd rather not block this PR if there's going to be a debate whether to support pageable.

Reply:
I would not block a preview based on this.